docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+# The autorenew feature
+
+## Goal
+-	Allow Hedera Services to automatically search for entities such as accounts, files, smart contracts, topics, tokens... that are about to expire and renew them.
+
+## Design
+- Introduce new settings in `application.properties`:
+  * `autorenew.isEnabled`
+  * `autorenew.numberOfEntitiesToCheck`
+  * `autorenew.maxNumberOfEntitiesToRenew`
+  * `autorenew.gracePeriod`
+- Each Hedera entity has an `expirationTime` which is the effective consensus timestamp at (and after) which the entity is set to expire.
+- Each Hedera entity also has an `autoRenewAccount` which is the account to pay for the fee at renewal. This `autoRenewAccount` could be itself if the entity is a crypto account or an account associated with the entity when it was created.
+- When a Hedera entity is created, its initial lifetime is defined by its `autoRenewPeriod`. At its `expirationTime`, Hedera Services will try to extend an entity's lifetime by another `autoRenewPeriod` if the `autoRenewAccount` has enough balance to do so, or as much extension as the remaining balance permits.
```

Comment:
What happens when someone specifies an `expirationTime` but not an `autoRenewAccount`? Will the entity be automatically expired? Perhaps someone wants create a topic for a limited period of time and doesn't wanted to be charged indefinitely for keeping it, but can't be bothered to delete it.

Reply:
When an entity is created, the fees charge for its creation include the payment (in advance) for how long it stays in the system, which means an entity is created with its expiration time. Without an `autoRenewAccount` or an `autoRenewAccount` with a zero balance at the time the entity expires, it cannot be automatically renewed and therefore will be deleted from the system.

Reply:
Thanks for clarifying. I think the scenario of expiration without autorenewaccount should be explicitly called out in the doc.